packages/react/src/ReactElement.js

```diff
@@ -290,6 +292,11 @@ export function cloneAndReplaceKey(oldElement, newKey) {
  * See https://reactjs.org/docs/react-api.html#cloneelement
  */
 export function cloneElement(element, config, children) {
+  invariant(
+    !(element === null || element === undefined),
```

Comment:
We could just use a loose equality check here

```js
element != null
```

Reply:
Ah right yes, because the type conversion will catch undefined 👍

Reply:
@gaearon didn't know that. We still have tons of places where we use loose equality checks.

Reply:
@gaearon okay gotcha, reverted that.

@aweary I can look into switching some over in a PR maybe? That should be pretty straightforward and I would love the opportunity to get more familiar with the codebase.

Reply:
@nicolevy I don't know, I'm not sure it's worth it? It could be helpful to make the changes and then somehow benchmark them. I'd be interested in seeing a profile that shows the deopts and their impact as well. Making all the checks strict would increase the overall bundle size, so there are other costs to consider.